.github/workflows/server-tests.yml

```
@@ -0,0 +1,50 @@
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  server-tests:
+    name: server tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)